### PR TITLE
[INJIMOB-2571]: refactor storage class

### PR DESCRIPTION
**Pull request by Alka1703**

### CODE REVIEW

File: shared/storage.ts [hunk 1]
```diff
@@ -573,5 +285,20 @@ export async function isMinimumLimitForBackupRestorationReached() {
   // APIs:
   // 1. CloudStorage.stat(file, context)
   // 2. getUncompressedSize()
-  return await Storage.isMinimumLimitReached('minStorageRequired');
+  return await isMinimumLimitReached('minStorageRequired');
```
Comment (KiruthikaJeyashankar) [inline shared/storage.ts:288]: Storage.isMinimumLimitReached was explicitly stating that we are checking storage related limit
But post moving isMinimumLimitReached function outside Storage class this mapping that its related to storage is missing, can we rename function accordingly to state that this function checks storage limit?

Comment (swatigoel) [reply shared/storage.ts:288]: Agree with @KiruthikaJeyashankar

File: shared/storage.ts [hunk 2]
```diff
@@ -52,111 +46,14 @@ async function generateHmac(
...
-        );
-        await this.setItem(key, encryptedValue, encryptionKey);
-        return true;
-      });
-    } catch (error) {
-      console.error('Error while loading backup data ', error);
-      return error;
-    }
+  static backupData = async (encryptionKey: string) => {
+    return await backupData(encryptionKey)
```
Comment (swatigoel) [inline shared/storage.ts:51]: This backupData function here causes confusion. As per import it's exportData but by looking at this line, it looks like recursion here.
so better to rename exportData data function and then use here

File: shared/storage.ts [hunk 3]
```diff
@@ -52,111 +46,14 @@ async function generateHmac(
...
   };
+ 
+  static async restoreBackedUpData(data: string, encryptionKey: string): Promise<any> {
```
Comment (swatigoel) [inline shared/storage.ts:54]: no need to use word backedUp, we can keep it simple restoreBackupData

File: shared/storage.ts [hunk 4]
```diff
@@ -573,5 +285,20 @@ export async function isMinimumLimitForBackupRestorationReached() {
   // APIs:
   // 1. CloudStorage.stat(file, context)
   // 2. getUncompressedSize()
-  return await Storage.isMinimumLimitReached('minStorageRequired');
+  return await isMinimumLimitReached('minStorageRequired');
 }
+
+export async function  isMinimumLimitReached(limitInMB: string) {
```
Comment (swatigoel) [inline shared/storage.ts:291]: rename this function to isMinimumStorageLimitReached?

File: shared/backupUtils/restoreData.ts [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+      }),
+    );
+  }
+
+  /**
+   * Verify the restored credential
+   * @param vcData  Verifiable Credential from backup
+   * @param vcMetadata  VC metadata
+   */
+  async function verifyCredential(vcData: any, vcMetadata: any) {
```
Comment (swatigoel) [inline shared/backupUtils/restoreData.ts:199]: verifyCredential can we keep this function in util and use in all flows like download credential, restore

**Review by swatigoel — CHANGES_REQUESTED** [review]

File: shared/storage.ts (hunk 1 above)
Comment (Alka1703) [reply shared/storage.ts:288]: Renamed it to `isMinimumStorageLimitReached`

File: shared/storage.ts (hunk 2 above)
Comment (Alka1703) [reply shared/storage.ts:51]: Missed renaming here. Did it now.

File: shared/storage.ts (hunk 3 above)
Comment (Alka1703) [reply shared/storage.ts:54]: Sure, Updating it.

File: shared/backupUtils/restoreData.ts (hunk 5 above)
Comment (Alka1703) [reply shared/backupUtils/restoreData.ts]: Done

File: shared/Utils.ts [hunk 6]
```diff
@@ -53,4 +56,22 @@ export const parseJSON = (input: any) => {
 
 export const isNetworkError = (error: string) => {
   return error.includes(NETWORK_REQUEST_FAILED);
-};
\ No newline at end of file
+};
+
+export async function verifyCredentialData(
```
Comment (swatigoel) [inline shared/Utils.ts:61]: instead of this Utils, move it to Utils under OpenIdVCI.

**Review by swatigoel — CHANGES_REQUESTED** [review]

File: shared/Utils.ts (hunk 6 above)
Comment (Alka1703) [reply shared/Utils.ts]: Done

**Review by swatigoel — APPROVED** [review]

**Pull request merged**